src/core/raster/qgsrasterinterface.cpp

```diff
@@ -336,7 +336,7 @@ void QgsRasterInterface::initHistogram( QgsRasterHistogram &histogram,
   }
   QgsDebugMsgLevel( QStringLiteral( "theHistogram.width = %1 histogram.height = %2" ).arg( histogram.width ).arg( histogram.height ), 4 );
 
-  int myBinCount = binCount;
+  long myBinCount = binCount;
```

Comment:
perhaps use long long or qint64, since long will be 32 bit on Linux 32 bit, and Windows 32 & 64 bit.